location/lbs/activity/audioset/dataprocessing/audio_processing.py

```diff
@@ -0,0 +1,466 @@
+"""
+Script to process a csv file from AudioSet into a csv file with features.
+
+Each row of the inputted csv is parsed for the video_id, start_time_seconds,
+end_time_seconds, and labels
+The downloaded audios would be clipped into the labelled 10 seconds segments
+The audio segments are then processed using Librosa to extract features
+
+Usage
+Standalone script:
+python audio_processing.py --src_dir PATH_TO_INPUT_FILES --des_dir DEST_DIR PATH_TO_OUTPUT_FILES_--audioset_csv CSV FILE --labels LABELS --redo BOOLEAN TO REDOWNLOAD THE YOUTUBE FILES
+example:
+python audio_processing.py --src_dir location/lbs/activity/audioset/dataprocessing/example_src_dir --des_dir location/lbs/activity/audioset/dataprocessing/example_dest_dir_--audioset_csv balanced_train_segments --labels "Gunshot, gunfire" --False
+
+Build with bazel:
+bazel build location/lbs/activity/audioset/dataprocessing:audio_processing
+bazel-bin/location/lbs/activity/audioset/dataprocessing/audio_processing --src_dir location/lbs/activity/audioset/dataprocessing/example_src_dir --des_dir location/lbs/activity/audioset/dataprocessing/example_dest_dir_--audioset_csv balanced_train_segments --labels "Gunshot, gunfire" --False
+
+
+--src_dir is the path to the directory containing the input files
+(audioset csv file ontology.json)
+--dest_dir is the path to the directory containing the output files
+(output csv file)
+--labels included in the AudioSet dataset can be used (i.e "Gunshot, gunfire")
+
+Performance: to confirm the presence of, extract features from, and output to a
+    csv file, takes about ___minutes for around 20,000 samples.
+"""
+
+from __future__ import unicode_literals
+import youtube_dl
+import csv
+import json
+from pydub import AudioSegment
+import os
+import shutil
+from os import listdir
+from os.path import isfile, join
+import librosa
+from absl import app
+from absl import flags
+from absl import logging
+import datetime
+
+FLAGS = flags.FLAGS
+
+
+def setup_abseil():
+    """Sets up command line arguments using Abseil.
+
+    Defines --audioset_csv, --labels, --redo, --src_dir, and --dest_dir as flags
+    """
+    flags.DEFINE_string('audioset_csv', 'balanced_train_segments',
+                        'The csv file of a dataset provided by the audioset')
+    flags.DEFINE_multi_string('labels', 'Gunshot, gunfire',
+                              'The set of labels identified by the audioset to '
+                              'treat as positive')
```

Comment:
only positive?

Reply:
Yes. The labels flag is meant to allow the user to designate which labels to treat as positive examples. Any label outside of the set the user passes in will be treated as negative examples. So, if you were training for the sound of gunshots, you would pass in --labels "Gunshot, gunfire for the labels flag. Now any audio segments labelled as "Gunshot, gunfire" is now a positive example, and any audio segments not labelled as "Gunshot, gunfire" is a negative example. I meant for this to allow for generalization and the ability to pick which labels you want to train for with the same dataset.